src/multisketch.rs

```diff
@@ -141,6 +152,92 @@ impl MultiSketch {
+        self.kmer_lengths() == sketch2.kmer_lengths()
+            && self.sketch_size == sketch2.sketch_size
+            && self.get_hash_type() == sketch2.get_hash_type()
+    }
+
+    pub fn merge_sketches(&self, sketch2: &Self) -> Self {
+        let mut sketch1_names: HashSet<String> = HashSet::new();
+        for sketch in self.sketch_metadata.iter() {
+            let name = sketch.name().to_string();
+            println!("{}", name);
+            sketch1_names.insert(name);
+        }
+
+        // First metadata
+        let mut merged_metadata = self.sketch_metadata.clone();
+        for sketch in sketch2.sketch_metadata.iter() {
+            if sketch1_names.contains(sketch.name()) {
+                let sample_name = sketch.name();
+                panic!("{sample_name} seems to appear in both databases. Cannot merge sketches.");
+            }
+            else {
+                merged_metadata.push(sketch.clone());
+            }
+        }
+        println!("Hello I am here");
+        // then merge sketches, create new multisketch instance
+        let mut merged_sketch = Self::new(&mut merged_metadata, self.sketch_size, &self.kmer_lengths, self.hash_type.clone());
+        println!("Empty merged sketch: {}", merged_sketch.sketch_bins.len());
+        // Merge actual sketche infos
+        merged_sketch.sketch_bins = self.sketch_bins.clone();
+        println!("merged sketch with sketches1: {}", merged_sketch.sketch_bins.len());
+        for bin in &sketch2.sketch_bins {
+            merged_sketch.sketch_bins.push(*bin);
+        }
+        println!("merged sketch with sketches2: {}", merged_sketch.sketch_bins.len());
+        // Update infos
+        merged_sketch.bin_stride = self.bin_stride;
+        merged_sketch.kmer_stride = self.kmer_stride;
+        merged_sketch.sample_stride = self.sample_stride;
+
+        let mut merged_name_map = self.name_map.clone();
+        merged_name_map.reserve(sketch2.sketch_metadata.len());
+
+        // update hashmap
+        for sketch in sketch2.sketch_metadata.iter() {
+            if !merged_name_map.contains_key(sketch.name()) {
+                let new_index = sketch.get_index() + self.sketch_metadata.len();
+                merged_name_map.insert(sketch.name().to_string(), new_index);
+            }
+        }
+
+        // update index
+        for sketch in merged_sketch.sketch_metadata.iter_mut() {
+            let new_index = merged_name_map[sketch.name()];
+            sketch.set_index(new_index);
+        }
+        merged_sketch.name_map = merged_name_map;
+
+        
+        merged_sketch
+    }
+
+    pub fn save_multi_sketches(&self, file_prefix: &str) -> Result<(), Box<dyn Error>> {
+        
+        let data_filename = format!("{file_prefix}.skd");
+        let mut file = std::fs::File::create(&data_filename)?;
+        SketchArrayFile::write_sketch_data(&mut file, &self.sketch_bins)?;
+        Ok(())
+    }
```

Comment:
I'm not clear on why this function is needed – should it be in sketch_datafile.rs?